### PR TITLE
Validate filenames server-side

### PR DESCRIPTION
**Pull request by atsmith813**
## What type of PR is this? (check all applicable)
- [x] Feature

## Description
This PR validates filename lengths server-side. I'm going to circle back in another PR to remove the excessive Datadog logging and rescues.

## Related Tickets & Documents
https://github.com/thepracticaldev/dev.to/projects/9#card-36356085

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
create_article
create_comment
create_organization
create_podcast

## Added tests?
- [x] yes

## Added to documentation?
- [x] no documentation needed

eye_roll_gif

### CODE REVIEW

**Comment by atsmith813** [comment]
Yeah, I'll move the method calls inside the controller action instead of `before_action`.

**Comment by atsmith813** [comment]
I'm going to ignore codeclimate since this is just an interim solution anyways. In the (near?) future, these code chunks will be removed.

File: app/controllers/image_uploads_controller.rb [hunk 1]
```diff
@@ -46,6 +53,12 @@ def cloudinary_link(uploaders)
 
   private
 
+  def valid_filename?
+    images = Array.wrap(params.dig("image"))
+    invalid_image = images.detect { |image| long_filename?(image) }
```
Comment (citizen428) [inline app/controllers/image_uploads_controller.rb:58]: Ruby has `Enumerable#none?`, which I think reads quite nicely here (you then won't need the last line, as the result of `none?` is the boolean we want):

```suggestion
    images.none? { |image| long_filename?(image) }
```

File: app/controllers/concerns/image_uploads.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  def log_image_data_to_datadog
+    images = Array.wrap(
+      params.dig("user", "profile_image") ||
+      params.dig("podcast", "image") ||
+      params.dig("organization", "profile_image") ||
+      params["image"],
+    )
+
+    raise if images.empty?
```
Comment (rhymes) [inline app/controllers/concerns/image_uploads.rb:19]: +1 on not raising a RuntimeError here

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -44,6 +44,12 @@ def edit
   # PATCH/PUT /users/:id.:format
   def update
     set_tabs(params["user"]["tab"])
+
+    unless valid_filename?
+      render :edit, status: :bad_request
```
Comment (rhymes) [inline app/controllers/users_controller.rb:49]: maybe `:unprocessable_entity` here?

Comment (atsmith813) [reply app/controllers/users_controller.rb:49]: I thought about that as well. But I wanted to keep it consistent with how the `update` action was already responding with errors:

https://github.com/thepracticaldev/dev.to/blob/e45363d63e0ed5a28db74b5dfe64c03d0a06c15b/app/controllers/users_controller.rb#L72

File: app/controllers/image_uploads_controller.rb (hunk 1 above)
Comment (atsmith813) [reply app/controllers/image_uploads_controller.rb]: Love it - updated! Always game to be reminded of the nifty Ruby methods 💯. Thank you!

**Comment by atsmith813** [comment]
Force pushed master just to double-check specs after the Travis CI debacle #7486

**Pull request merged**